src/flaskapp/api.py

```diff
@@ -0,0 +1,226 @@
+from flask import Flask, request
+
+from src.users.user_profile import (
+    get_user_profile,
+    get_user_profiles,
+    create_user_profile,
+    update_user_profile,
+)
+
+from src.teams.team_profile import (
+    get_team_profile,
+    get_team_profiles,
+    create_team_profile,
+    update_team_profile,
+)
+from src.teams.team_complete import team_complete
+from src.teams.user_leave import user_leave
+
+from src.teams.unify.team_invite import team_invite
+from src.teams.unify.team_confirm import team_confirm
+from src.teams.unify.team_rescind import team_rescind
+from src.teams.unify.team_reject import team_reject
+
+# from src.matching.team_recommendations import get_team_recommendations
+
+from src.flaskapp.util import format_string
+from src.flaskapp.schemas import ensure_user_logged_in
+
+app = Flask(__name__)
+
+@app.route("/", methods=["GET"])
+def index():
+    return {"message": "Welcome to TeamRU!"}, 200
+
+############################## USERS ##############################
+
+
+@app.route("/users", methods=["GET", "POST"])
+@ensure_user_logged_in
+def users(email):
+    email = None  # unused
+
+    if request.method == "GET":
+        # Filter response using query parameters
+        # Might need to add pagination (limit/offset) for this response
+        return get_user_profiles(request.args)
+
+    if request.method == "POST":
+        # Create a new user
+        data = request.get_json(silent=True)
+        email = format_string(data["user_email"])
+
+        prizes = []
+        skills = []
+        interests = []
+        bio = ""
+        github = ""
+        seriousness = -1  # -1 if users doesn't have one
+
+        if "prizes" in data:
+            prizes = format_string(data["prizes"])
+        if "skills" in data:
+            skills = format_string(data["skills"])
+        if "interests" in data:
+            interests = format_string(data["interests"])
+        if "bio" in data:
+            bio = format_string(data["bio"])
+        if "github" in data:
+            # NOTE can ping github api to verify this is an actual acct.
+            github = format_string(data["github"])
+        if "seriousness" in data:
+            try:
+                seriousness = int(data["seriousness"])
+            except ValueError:
+                seriousness = -1
+        return create_user_profile(
+            email,
+            prizes=prizes,
+            skills=skills,
+            bio=bio,
+            github=github,
+            interests=interests,
+            seriousness=seriousness,
+        )
+
+
+@app.route("/users/profile", methods=["GET", "PUT"])
```

Comment:
can we just do `/users/<user_id>` so that it matches how we are doing it with teams?

Reply:
or did we do this bc of JWT?

Reply:
Since we already get the user's email from their JWT, I figured it would be redundant to require the endpoint to also require a `user_id`. Also removes our need for base64 encoding/decoding entirely. That being said, I do agree that standardization is useful - we could do something where require email in the path, and then verify that the email from the token matches the email in the path. What do you think?

Reply:
Yea we can keep it like this then.